pygments/lexers/templates.py

```diff
@@ -1251,7 +1251,8 @@ class HtmlDjangoLexer(DelegatingLexer):
 
     name = 'HTML+Django/Jinja'
     aliases = ['html+django', 'html+jinja', 'htmldjango']
-    alias_filenames = ['*.html', '*.htm', '*.xhtml']
+    filenames = ['*.html.j2', '*.htm.j2', '*.xhtml.j2', '*.html.jinja2', '*.htm.jinja2', '*.xhtml.jinja2']
+    alias_filenames = ['*.html.j2', '*.htm.j2', '*.xhtml.j2', '*.html.jinja2', '*.htm.jinja2', '*.xhtml.jinja2']
```

Comment:
- Why the duplication between `filenames` and `alias_filenames`?
- Do you really want to remove the `*.html` pattern? Right now, it tries to guess a lexer with `analyse_text` methods. This will make all `*.html` files highlighted by the `HtmlLexer` exclusively.